src/components/query-panel/QueryPanelCtrl.ts

```diff
@@ -0,0 +1,84 @@
+import CodeMirror = require("codemirror");
+import {RestService} from "../../services/rest/RestService";
+import {EditorConfiguration} from "codemirror";
+import {isNotNullOrUndefined, isObject, isEmptyString} from "../../common/utils/Utils";
+
+export class QueryPanelCtrl implements ng.IComponentController {
+  static $inject: string[] = ["restService", "$timeout"];
+
+  cacheName: string;
+  name: string;
+  description: string;
+  statusCode: number;
+  editor: CodeMirror.Editor;
+  panelType:string;
+  inputParams: string [];
```

Comment:
Do you intend to have more than two inputs? If so an object would probably be a better choice than an array as we can then add meaningful keys to go with the stored value. If not then two appropriately named variables would suffice.

Reply:
Not sure at this point, maybe leave them for now until we see what the final number of params is?

Reply:
Sure